src/cmd/singularity/cli/sif.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/singularityware/singularity/docs"
+	"github.com/singularityware/singularity/src/pkg/buildcfg"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+	"github.com/spf13/cobra"
+)
+
+var (
+	deffile  string
+	partfile string
+	content  string
+	fstype   string
+	parttype string
+	uuid     string
+)
+
+func init() {
+	SifCmd.Flags().SetInterspersed(false)
+	SingularityCmd.AddCommand(SifCmd)
+	// Create
+	SifCmd.AddCommand(SifCreate)
+	SifCreate.Flags().StringVarP(&deffile, "deffile", "D", "", "include definitions file 'deffile'")
+	SifCreate.Flags().StringVarP(&partfile, "partfile", "P", "", "include file system partition `partfile'")
+	SifCreate.Flags().StringVarP(&content, "CONTENT", "c", "", "freeform partition content string")
+	SifCreate.Flags().StringVarP(&fstype, "FSTYPE", "f", "", "filesystem type: EXT3, SQUASHFS")
+	SifCreate.Flags().StringVarP(&parttype, "PARTTYPE", "p", "", "filesystem partition type: SYSTEM, DATA, OVERLAY")
+	SifCreate.Flags().StringVarP(&uuid, "uuid", "u", "", "pass a uuid to use instead of generating a new one")
+	// List
+	SifCmd.AddCommand(SifList)
+	// Dump
+	SifCmd.AddCommand(SifDump)
+	// Header ifHeader
+	SifCmd.AddCommand(SifHeader)
+	// Info
+	SifCmd.AddCommand(SifInfo)
+	// Del
+	SifCmd.AddCommand(SifDel)
+}
+
+var sif = buildcfg.SBINDIR + "/sif"
+
+// SifCmd represent the sif CLI cmd
+var SifCmd = &cobra.Command{
+	Use:  docs.SifUse,
+	Args: cobra.MinimumNArgs(1),
+	Run:  nil,
+}
+
+// SifCreate sif create cmd
+var SifCreate = &cobra.Command{
+	Use:     docs.SifCreateUse,
+	Short:   docs.SifCreateShort,
+	Example: docs.SifCreateExample,
+	Args:    cobra.MinimumNArgs(1),
```

Comment:
minimally, sif create requires at least : sif create -P -c -f -p container.sif which seems like 5 arguments not 1.

Reply:
I think `Args` is the number of _positional_ arguments.

Ref: https://godoc.org/github.com/spf13/cobra#PositionalArgs

Arguably (heh), this should be using `cobra.ExactArgs` instead of `cobra.MinimumNArgs` though, since exactly one file must be specified?

Reply:
you are right @tri-adam 
> `Args` the number of positional arguments.